Added ability to customize time scale tick marks formatter (#386)
tradingview/lightweight-charts: docs/time-scale.md

|`visible`|`boolean`|`true`|If true, the time scale is shown on a chart|
|`timeVisible`|`boolean`|`false`|If true, the time is shown on the time scale and in the vertical crosshair label|
|`secondsVisible`|`boolean`|`true`|If true, seconds are shown on the label of the crosshair vertical line in `hh:mm:ss` format on intraday intervals|
|`tickMarkFormatter`|`(TimePoint, TickMarkType, locale) => string`|Default tick marks formatter|Allows override tick marks formatter (see below)|

### Tick marks formatter

Tick marks formatter might be used to customize tick marks labels on time axis.

To customize it, you need to provide `tickMarkFormatter` option. It's the function with the following declaration:

```typescript
export type TickMarkFormatter = (timePoint: TimePoint, tickMarkType: TickMarkType, locale: string) => string;
```

Where `timePoint` is [Time](./time.md) object, `type` is [TickMarkType](./constants.md#TickMarkType) enum and `locale` is a string with current applied locale.

> This function should return a string represents `timePoint` as `tickMarkType` type (year, month, etc) in `locale` locale.

Reviewer: ```suggestion
This function should return `timePoint` as a string formatted according to `tickMarkType` type (year, month, etc) and `locale`.
```

> Note that the returned string should be shortest value as possible and contains no more than (approx.) 8 characters.

Reviewer: ```suggestion
Note that the returned string should be the shortest possible value and should have no more than 8 characters.
```

> Otherwise, tick marks could be overlap each other.

Reviewer: ```suggestion
Otherwise, the tick marks will overlap each other.
```
